src/metabase/models/revision/diff.clj

```diff
@@ -88,6 +88,9 @@
     [:result_metadata _ _]
     (deferred-tru "edited the metadata")
 
+    [:metabase_version _ _]
```

Comment:
Should we add a test for this in `metabase.models.card-test/diff-cards-str-test`?

Reply:
Why do we want a revision string for changing the metabase version? Metabase versions are not something that the user has control over.

Reply:
Is this shown to a user? I'm not sure, but could it be useful for debugging purposes, i.e. "right here you've upgrade your metabase"? I'm not totally sure where is this string shown.

Reply:
These strings are shown to users in the revision history of a question. Here's how they're displayed:
https://www.loom.com/share/162c4a0b48df4898bba9a5153066fa5e?sid=37c740b6-bfd8-4754-8deb-44956e5fb17d

Reply:
The way we decide whether to store a new revision or not is to compare serialized objects. 
https://github.com/metabase/metabase/blob/c9bd641a7751e22f181f1e1b6be905854eb4c272/src/metabase/models/revision.clj#L179-L180

If `:metabase_version` is included in the serialized object (which we want) then changing the `:metabase_version` will trigger another revision to be stored even if nothing else changed. That's not what we want though, because each revision is meant to represent things users have done, not migrations we have made. So I'd say the solution here is to `dissoc` `:metabase_version` from the serialized objects in the comparison.

Reply:
So the logic is the following: report_card stores metabase version used to _create_ that card (and is never updated), and revisions store respective metabase version.

Now that I think about it... maybe it makes sense to add `metabase_version` table to revisions globally (rather than only for cards in their properties)? Like, for audit purposes, every revision will know when it was created.

Reply:
> maybe it makes sense to add metabase_version table to revisions globally

Good point, if it makes sense for cards it does for other types of revisioned objects, e.g. dashboards.

> So the logic is the following: report_card stores metabase version used to create that card (and is never updated), and revisions store respective metabase version.

Oh, I just noticed that. That's unhelpful information by itself imo, and I find it confusing to have the same key represent two different things depending on whether the object is stored in a revision object or in the actual table. It's unhelpful because a card can be updated in subsequent versions after it was created, which means any corrupt data we find might be due to those versions it was updated in, and not the version that card was created. In order to debug problems with the data in general, we'd have to know the full revision history and what versions it was updated in. I'd recommend going back to @paoliniluis and confirming if we really only want the creation version on report_card.

Reply:
That's what @likeshumidity actually requested, because initially I did update `metabase_version` all the time...

Reply:
Fair enough then, I didn't see that. He must have his reasons. Anyway, that doesn't change the problem I mentioned above https://github.com/metabase/metabase/pull/34520#discussion_r1358167544.

Reply:
Yep, so I'm removing `metabase_version` from a serialized object, but will add it as a field to a `revision` table. So that all revisions will know which version of Metabase was used to create them (as I finally understood #29167 requests).

This will also remove a need for `metabase_version` in diffs. :)

Reply:
Here's a loom if what I said above was unclear: https://www.loom.com/share/0d72376a08564867a11c773f463f221a?sid=1e563f2e-83fe-44df-b1bc-06a62991ecc5

Reply:
So I've updated code to separate store metabase_version for cards (and not store that in revisions because it would be just a noise I think, if it's not changing), and separately for revisions.

Also, thanks for that Loom! I now understood what you've meant, but thankfully last changes circumvent this. :)